api/src/main/java/jakarta/data/page/Pagination.java

```diff
@@ -131,6 +131,16 @@ public PageRequest<T> previous() {
         }
     }
 
+    @Override
+    public PageRequest<T> nextWithKey(Object... key) {
+        return new Pagination<>(page + 1, this.size, this.sorts, Mode.OFFSET, Cursor.forKey(key), requestTotal);
+    }
+
+    @Override
+    public PageRequest<T> previousWithKey(Object... key) {
+        return page()<=1 ? null : new Pagination<>(page - 1, this.size, this.sorts, Mode.OFFSET, Cursor.forKey(key), requestTotal);
+    }
```

Comment:
Because cursor-based pagination allows for updates to the data which can cause new entries to appear, I think we have language in the spec that says that a previous page to page 1 should also be numbered as 1.
The JavaDoc to the new method you proposed doesn't state anything about the possibility of a null result.
```suggestion
    public PageRequest<T> previousWithKey(Object... key) {
        return new Pagination<>(page > 1 ? page - 1 : 1, this.size, this.sorts, Mode.OFFSET, Cursor.forKey(key), requestTotal);
    }
```

Reply:
>  I think we have language in the spec that says that a previous page to page 1 should also be numbered as 1.

OK, I missed that, but I think it's fine.